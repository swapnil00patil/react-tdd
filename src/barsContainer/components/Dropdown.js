import React from 'react'
import styled from 'styled-components'

const Select = styled.select`
  height: 33px;
  width: 150px;
  font-size: 18px;
  background: transparent;
  cursor:pointer;
`
const Option = styled.option`

`

class Dropdown extends React.Component {
  shouldComponentUpdate (nextProps) {
    if (this.props.bars && nextProps.bars.length === this.props.bars.length) {
      return false
    } else {
      return true
    }
  }

  render () {
    const { bars, selectBar, className } = this.props
    return (
      <Select onChange={selectBar} className={className}>
        {bars.map((value, index) => <Option key={index} value={value}>#progress{ index + 1}</Option>)}
      </Select>
    )
  }
}

export default Dropdown
