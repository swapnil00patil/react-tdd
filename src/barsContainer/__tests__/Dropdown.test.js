import React from 'react'
import { shallow, mount } from 'enzyme'

import Dropdown from '../components/Dropdown'

describe('Dropdown', () => {
  it('Dropdown should be defined', () => {
    expect(Dropdown).toBeDefined()
  })

  it('should render exactly same number bars', () => {
    const wrapper = mount(<Dropdown bars={[12, 13, 14]} />)
    wrapper.find('option').length.valueOf(3)
  })

  it('check correct value is retrieved when change dropdown', () => {
    const onChange = jest.fn()
    const wrapper = mount(<Dropdown bars={[12, 13, 14]} selectBar={onChange} />)
    wrapper.first('select').simulate('change', {})

    expect(onChange).toHaveBeenCalled()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<Dropdown bars={[12, 13, 14]} />)
    expect(wrapper).toMatchSnapshot()
  })
})
