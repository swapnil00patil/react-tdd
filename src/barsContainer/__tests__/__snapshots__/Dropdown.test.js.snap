// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    bars={
        Array [
            12,
            13,
            14,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.option
          value={12}
>
          #progress
          1
</styled.option>,
        <styled.option
          value={13}
>
          #progress
          2
</styled.option>,
        <styled.option
          value={14}
>
          #progress
          3
</styled.option>,
      ],
      "className": undefined,
      "onChange": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "#progress",
            1,
          ],
          "value": 12,
        },
        "ref": null,
        "rendered": Array [
          "#progress",
          1,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "#progress",
            2,
          ],
          "value": 13,
        },
        "ref": null,
        "rendered": Array [
          "#progress",
          2,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "#progress",
            3,
          ],
          "value": 14,
        },
        "ref": null,
        "rendered": Array [
          "#progress",
          3,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.option
            value={12}
>
            #progress
            1
</styled.option>,
          <styled.option
            value={13}
>
            #progress
            2
</styled.option>,
          <styled.option
            value={14}
>
            #progress
            3
</styled.option>,
        ],
        "className": undefined,
        "onChange": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "#progress",
              1,
            ],
            "value": 12,
          },
          "ref": null,
          "rendered": Array [
            "#progress",
            1,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "#progress",
              2,
            ],
            "value": 13,
          },
          "ref": null,
          "rendered": Array [
            "#progress",
            2,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "#progress",
              3,
            ],
            "value": 14,
          },
          "ref": null,
          "rendered": Array [
            "#progress",
            3,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
